test/e2e/mock-e2e.js

```diff
@@ -138,37 +138,48 @@ const privateHostMatchers = [
  * @param {object} options - Network mock options.
  * @param {string} options.chainId - The chain ID used by the default configured network.
  * @param {string} options.ethConversionInUsd - The USD conversion rate for ETH.
+ * @param {boolean} options.disableMocking - Whether to disable mocking and allow all requests to pass through.
  * @returns {Promise<SetupMockReturn>}
  */
 async function setupMocking(
   server,
   testSpecificMock,
-  { chainId, ethConversionInUsd = 1700 },
+  { chainId, ethConversionInUsd = 1700, disableMocking = false },
```

Comment:
mm instead of adding the condition here, why don't we don't we check it at the helpers level, so if there's disableMocking passed from a spec, we simply don't start the server (neither stop it).
Then we don't need to modify this file at all, and we are not starting a background server for nothing, if that makes sense